providers/openmetrics/client_metrics.go

```diff
@@ -0,0 +1,168 @@
+				Name: "grpc_client_msg_received_total",
+				Help: "Total number of RPC stream messages received by the client.",
+			}), []string{"grpc_type", "grpc_service", "grpc_method"}),
+
+		clientStreamMsgSent: prom.NewCounterVec(
+			opts.apply(prom.CounterOpts{
+				Name: "grpc_client_msg_sent_total",
+				Help: "Total number of gRPC stream messages sent by the client.",
+			}), []string{"grpc_type", "grpc_service", "grpc_method"}),
+
+		clientHandledHistogramEnabled: false,
+		clientHandledHistogramOpts: prom.HistogramOpts{
+			Name:    "grpc_client_handling_seconds",
+			Help:    "Histogram of response latency (seconds) of the gRPC until it is finished by the application.",
+			Buckets: prom.DefBuckets,
+		},
+		clientHandledHistogram:           nil,
+		clientStreamRecvHistogramEnabled: false,
+		clientStreamRecvHistogramOpts: prom.HistogramOpts{
+			Name:    "grpc_client_msg_recv_handling_seconds",
+			Help:    "Histogram of response latency (seconds) of the gRPC single message receive.",
+			Buckets: prom.DefBuckets,
+		},
+		clientStreamRecvHistogram:        nil,
+		clientStreamSendHistogramEnabled: false,
+		clientStreamSendHistogramOpts: prom.HistogramOpts{
+			Name:    "grpc_client_msg_send_handling_seconds",
+			Help:    "Histogram of response latency (seconds) of the gRPC single message send.",
+			Buckets: prom.DefBuckets,
+		},
+		clientStreamSendHistogram: nil,
+	}
+}
+
+// Describe sends the super-set of all possible descriptors of metrics
+// collected by this Collector to the provided channel and returns once
+// the last descriptor has been sent.
+func (m *ClientMetrics) Describe(ch chan<- *prom.Desc) {
+	m.clientStartedCounter.Describe(ch)
+	m.clientHandledCounter.Describe(ch)
+	m.clientStreamMsgReceived.Describe(ch)
+	m.clientStreamMsgSent.Describe(ch)
+	if m.clientHandledHistogramEnabled {
+		m.clientHandledHistogram.Describe(ch)
+	}
+	if m.clientStreamRecvHistogramEnabled {
+		m.clientStreamRecvHistogram.Describe(ch)
+	}
+	if m.clientStreamSendHistogramEnabled {
+		m.clientStreamSendHistogram.Describe(ch)
+	}
+}
+
+// Collect is called by the Prometheus registry when collecting
+// metrics. The implementation sends each collected metric via the
+// provided channel and returns once the last metric has been sent.
+func (m *ClientMetrics) Collect(ch chan<- prom.Metric) {
+	m.clientStartedCounter.Collect(ch)
+	m.clientHandledCounter.Collect(ch)
+	m.clientStreamMsgReceived.Collect(ch)
+	m.clientStreamMsgSent.Collect(ch)
+	if m.clientHandledHistogramEnabled {
+		m.clientHandledHistogram.Collect(ch)
+	}
+	if m.clientStreamRecvHistogramEnabled {
+		m.clientStreamRecvHistogram.Collect(ch)
+	}
+	if m.clientStreamSendHistogramEnabled {
+		m.clientStreamSendHistogram.Collect(ch)
+	}
+}
+
+// EnableClientHandlingTimeHistogram turns on recording of handling time of RPCs.
+// Histogram metrics can be very expensive for Prometheus to retain and query.
+func (m *ClientMetrics) EnableClientHandlingTimeHistogram(opts ...HistogramOption) {
```

Comment:
@bwplotka How should we enable the options? Much similar to logging middlewares, or do you have something different in mind? 🤔

Reply:
I have enabled the previous convention to enable histogram options. Let me know if the changes make sense